lib/broadway_rabbitmq/producer.ex

```diff
@@ -77,11 +83,40 @@ defmodule BroadwayRabbitmq.Producer do
   end
 
   @impl true
+  def terminate(_reason, state) do
+    %{client: client, channel: channel} = state
+
+    if channel do
+      client.close_connection(channel.conn)
```

Comment:
Maybe the client should monitor this process instead of relying on terminate to be invoked?